app/parsers/bulkrax/application_parser.rb

```diff
@@ -242,15 +242,48 @@ def file_sets_total
     def write
       write_files
       zip
+      # uncomment next line to debug for faulty zipping during bagit export
+      bagit_zip_file_size_check if importerexporter.parser_klass.include?('Bagit')
     end
 
     def unzip(file_to_unzip)
-      WillowSword::ZipPackage.new(file_to_unzip, importer_unzip_path).unzip_file
+      Zip::File.open(file_to_unzip) do |zip_file|
+        zip_file.each do |entry|
+          entry_path = File.join(importer_unzip_path, entry.name)
+          FileUtils.mkdir_p(File.dirname(entry_path))
+          zip_file.extract(entry, entry_path) unless File.exist?(entry_path)
+        end
+      end
     end
 
     def zip
       FileUtils.rm_rf(exporter_export_zip_path)
-      WillowSword::ZipPackage.new(exporter_export_path, exporter_export_zip_path).create_zip
+      Zip::File.open(exporter_export_zip_path, create: true) do |zip_file|
+        Dir["#{exporter_export_path}/**/**"].each do |file|
+          zip_file.add(file.sub("#{exporter_export_path}/", ''), file)
+        end
+      end
+    end
+
+    # TODO: remove Entry::BagitZipError as well as this method when we're sure it's not needed
```

Comment:
Can we create a ticket to track / not forget about this?

Reply:
Will do!